receiver/prometheusreceiver/metrics_receiver.go

```diff
@@ -92,36 +110,51 @@ func (r *pReceiver) Start(_ context.Context, host component.Host) error {
 
 	allocConf := r.cfg.TargetAllocator
 	if allocConf != nil {
-		go func() {
-			// immediately sync jobs and not wait for the first tick
-			savedHash, _ := r.syncTargetAllocator(uint64(0), allocConf, baseCfg)
-			r.targetAllocatorIntervalTicker = time.NewTicker(allocConf.Interval)
-			for {
-				<-r.targetAllocatorIntervalTicker.C
-				hash, err := r.syncTargetAllocator(savedHash, allocConf, baseCfg)
-				if err != nil {
-					r.settings.Logger.Error(err.Error())
-					continue
-				}
-				savedHash = hash
-			}
-		}()
+		err = r.initTargetAllocator(allocConf, baseCfg)
+		if err != nil {
+			return err
+		}
 	}
 
+	r.reloadReady.Close()
+
 	return nil
 }
 
+func (r *pReceiver) initTargetAllocator(allocConf *targetAllocator, baseCfg *config.Config) error {
+	r.settings.Logger.Info("Starting target allocator discovery")
+	// immediately sync jobs and not wait for the first tick
+	savedHash, err := r.syncTargetAllocator(uint64(0), allocConf, baseCfg)
+	if err != nil {
+		return err
+	}
+	r.targetAllocatorIntervalTicker = time.NewTicker(allocConf.Interval)
+	go func() {
+		<-r.reloadReady.C
```

Comment:
Wondering what is writing to this channel that we're reading from here?

Reply:
We're closing the channel after all the components have been initialized [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/14464/files/a46743edad8c4c67e2a19cb67722966d252c3ade#diff-8172ccc732b059ffa8d9d8858643701e32c90c3d3ab8b6697fa74ca1c8c09fb5R119)